components/script/dom/readablestreamdefaultcontroller.rs

```diff
@@ -28,8 +29,9 @@ use crate::script_runtime::{JSContext, JSContext as SafeJSContext};
 /// The fulfillment handler for
 /// <https://streams.spec.whatwg.org/#readable-stream-default-controller-call-pull-if-needed>
 #[derive(Clone, JSTraceable, MallocSizeOf)]
+#[allow(crown::unrooted_must_root)]
 struct PullAlgorithmFulfillmentHandler {
-    controller: DomRoot<ReadableStreamDefaultController>,
+    controller: Dom<ReadableStreamDefaultController>,
```

Comment:
I don't think this is safe, since these structs aren't visible to the GC. That's why I recommend Trusted here.

Reply:
> That's why I recommend Trusted here.
I just tried it and it breaks `Callback: JSTraceable + MallocSizeOf `. 

>  these structs aren't visible to the GC

But they end-up on the `PromiseNativeHandler`, which is a dom_struct.

Reply:
I've strengthened this by keeping the controller rooted while the native promise handler is setup.